src/bpftrace.cpp

```diff
@@ -1943,14 +1943,17 @@ std::unordered_set<std::string> BPFtrace::get_raw_tracepoint_modules(
   return mod != rts.end() ? mod->second : std::unordered_set<std::string>();
 }
 
-const struct stat &BPFtrace::get_pidns_self_stat() const
+const std::optional<struct stat> &BPFtrace::get_pidns_self_stat() const
```

Comment:
I think it's more important that the value inside the optional is const rather than the optional itself, which still means a caller could still modify the stat struct

Reply:
damn i've been writing too much rust huh

Reply:
According to this: https://stackoverflow.com/questions/55060223/is-the-content-of-a-const-stdoptional-always-const
(which I did not test) I think what we have here is what we want. `std::optional<const struct stat>` would allow wholesale replacement of the value inside the optional.

Reply:
@danobi why do we want to protect the optional instead of the thing inside it which was (prior to this change) a const ref?

Reply:
Const on the outside protects both the optional and the contained value. Const on the inside does not protect the optional - I think it is strictly worse.

Reply:
Ugh, you're right. I read that overflow answer too fast. Sorry for the noise.